text/0000-at-least-one-feature.md

```diff
@@ -191,14 +192,89 @@ there is no point of adding such a new minimal feature because there is nothing
 This solution is more mathematically elegant, but it seems harder to implement.
 It is unclear how Cargo could require the Rust code to obey this property without new infra like the portability lint.
 
+Finally, in a sense these variations are actually isomorphic.
+If no one can request the empty feature set, no one can observe what library interface that corresponds to.
+There is no reason to conclude that it *isn't* the empty library.
+
+## RFC [#3146](https://github.com/rust-lang/rfcs/pull/3146)
+
+This more more expensive solution aimed to solve the "running out of good names" problem this has.
+
+Returning to our commutative diagram, instead of
+
+- removing the ⊥ feature downset (this proposal)
+
+- mandating the ⊥ feature downset correspond to ⊥ the empty library interface (first alternative, extra rules on the vertical mappings in diagrams)
+
+We can instead
+
+- allow customizing the mapping from old feature names to new feature names (customizing top horizontal mapping)
+
+This works and allows giving things the "right" names" every time, but is much work work to implement.
+
+I think it might make sense to do this someday, but it is better to start with this.
+My thinking about this is basically "walk before run".
+Today, that would be lots of work for an obscure problem, and a solution that downstream crates need to know about because it affects how dependencies are written.
+However if we do this first, "opt-out features" are poised to become much more popular now that they are no longer a compatibility hazard.
+We can evaluate both that new popularity, and the incidence of the "bad name" problem in *practice*, and use that to justify a more heavyweight solutions.
+
+## RFC [#3283](https://github.com/rust-lang/rfcs/pull/3283)
+
+This proposal adds a new notion of a "feature base", which aims to essential provide more `default-features = false` options rather than just one forever.
+`default-features = false` is ret-conned as the first feature base.
+
+My view is that that RFC is pretty close to being this one in disguise.
+A "feature base" is just a feature.
+The fact that there is always one "feature base" is just another way of accomplishing the "at least one feature"
+restriction.
+
+The main difference is that `default-features = false` is ret-conned as a feature to allow for existing users.
+That can be thought of from the perspective of this RFC as a "one-off" feature migration to get reverse dependencies predating the use of `at-least-one-feature = true` on to a feature so they retroactively abide by the "at least one feature" rule.
+
+## "Negative features"
+
+I take a pretty dim view of this.
```

Comment:
Unsure how its normally done but I'd recommend writing these RFCs from the Rust-community perspective, rather than from your own personal perspective, considering the community will be accepting or rejecting this and it will be viewed as the voice of community, with little or no knowledge of the author.

So let it stand on its merits rather than making a judgement.

Reply:
I feel like it is hard to do this while the RFC is provisional, some mix of facts and opinions is unavoidable. But if this gets close to concensus I am happy to go remove all the "I"s and rewrite according to what the team consensus position actually is.

Reply:
I did strike the first sentence however. The next one is fine enough as a topic sentence for now.